pkg/apis/batch/validation/validation.go

```diff
@@ -55,16 +55,16 @@ const (
 )
 
 var (
-	supportedPodFailurePolicyActions sets.String = sets.NewString(
+	supportedPodFailurePolicyActions = sets.New(
```

Comment:
if you change the implementation of `NotSupported` like this:

```diff
--- a/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go
+++ b/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go
@@ -200,12 +200,12 @@ func Invalid(field *Path, value interface{}, detail string) *Error {
 // NotSupported returns a *Error indicating "unsupported value".
 // This is used to report unknown values for enumerated fields (e.g. a list of
 // valid values).
-func NotSupported(field *Path, value interface{}, validValues []string) *Error {
+func NotSupported[S ~string](field *Path, value interface{}, validValues []S) *Error {
        detail := ""
        if len(validValues) > 0 {
                quotedValues := make([]string, len(validValues))
                for i, v := range validValues {
-                       quotedValues[i] = strconv.Quote(v)
+                       quotedValues[i] = strconv.Quote(string(v))
                }
                detail = "supported values: " + strings.Join(quotedValues, ", ")
        }
```

Then we won't need to convert everything into string here.